consensus/consensus-types/src/proposal_msg.rs

```diff
@@ -168,13 +161,15 @@ impl<T: Payload> fmt::Display for ProposalMsg<T> {
     }
 }
 
-impl<T: Payload> TryFrom<ProposalMsg<T>> for network::proto::Proposal {
+impl<T: Payload> TryFrom<ProposalMsg<T>> for network::proto::Proposal
+where
+    T: Serialize + Default + PartialEq,
```

Comment:
why do you need this? Payload includes everything

Reply:
Oh, that's nice, let me try without this :)